.github/workflows/imperative.yml

```diff
@@ -0,0 +1,75 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Imperative CI
+
+on: [push]
+
+jobs:
+  test:
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [8.x, 10.x, 12.x]
+        os: [windows-latest, ubuntu-latest, macos-latest]
+
+    env:
+      OS: ${{ matrix.os }}
+      NODE: ${{ matrix.node-version }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Install Node Package Dependencies
+      id: install
+      run: npm ci
+
+    - name: Build Source
+      id: build
+      run: npm run build --if-present
+
+    - name: Unit Tests
+      id: unit
+      if: ${{ always() && steps.build.outcome == 'success' }}
+      run: npm run test:unit >> file.txt
+
+    - name: Unlock Keyring
+      if: ${{ always() && steps.build.outcome == 'success' && matrix.os == 'ubuntu-latest' }}
+      uses: tjohnsonBCM/unlock-keyring@master
+
+    - name: Integration Tests
+      id: integration
+      if: ${{ always() && steps.build.outcome == 'success' }}
+      run: npm run test:integration >> file.txt
+
+    - name: Zip Results
```

Comment:
I think this stage of zipping the test results could be removed. The GitHub workflow for Zowe Explorer uploads the folder directly, and it is automatically zipped.
```yaml
- name: Upload test results
  uses: actions/upload-artifact@v1
  with:
    name: results
    path: results/
```

This might also resolve the issue we had with artifacts being doubly zipped 🙂

Reply:
I took this action, but just want you to be aware why it was being done. Archiving artifacts does indeed put them in a zip file, but they are being zipped without compression. Therefore, with this change, the archived artifacts swell in size from 3.5MB per build (~31.5MB total) to 25MB per build (~225MB total).

Reply:
@awharn Sorry I think you mentioned this before and I forgot :cry: Interesting that GitHub creates a ZIP without compressing. Considering the compression benefits we were getting from the separate ZIP step, I'm OK with either method (using the ZIP step or not).

Reply:
I have no real preference. AFAIK there isn't a storage limit on public repos. If it were private, I think we'd only get 9 runs out of the thing before we ran out of space. The only real argument I can make is download time, but 25 Megabytes shouldn't be too bad.

Reply:
Not sure if the 25 MB indicates how much space is actually being used on GitHub's servers, or if GitHub remembers how big the folder was before compressing it. I see this in web browser:
![image](https://user-images.githubusercontent.com/50887074/88178670-bbb8cd80-cbf8-11ea-9297-49c8ede8e4f9.png)
But when I download the ZIP, I see this in Windows Explorer:
![image](https://user-images.githubusercontent.com/50887074/88178771-e571f480-cbf8-11ea-963f-94553d8919f0.png)

Reply:
Oh. Hm. Well, if that is the case, then we can just get rid of the zip step. But I think it is odd it displays the uncompressed size on the Web UI right next to the download button.